public/app/features/alerting/unified/hooks/rules/usePendingPeriod.tsx

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react';
+
+import { CombinedRule } from 'app/types/unified-alerting';
+
+import { isRecordingRulerRule, isRecordingRule, isAlertingRulerRule, isAlertingRule } from '../../utils/rules';
+import { formatPrometheusDuration } from '../../utils/time';
+
+export function usePendingPeriod(rule: CombinedRule): string | undefined {
```

Comment:
not important but, in terms of reusing the logic, maybe we could consider not having a hook, and just keep the logic without the useMemo?  Consumers of this method can use this method without the restriction of being a component or hook.

Reply:
Good point

Reply:
I'll extract the function but keep the hook, as we already use it in two components